### PR TITLE
Fix: Add user profile with dropdown after sign in through github

### LINKED ISSUE #530: Profile Icon and Name Not Displayed After GitHub Sign-In
**Issue Description:**
When attempting to sign in with GitHub on dashboard.realdevsquad.com, the sign-in process is successful; however, the user's profile icon and name are not displayed, and the "Sign In" option persists. This issue prevents users from accessing their profiles and the associated functionality. 

**Steps to Reproduce:**
1. Go to dashboard.realdevsquad.com.
2. Click on the "Sign In with GitHub" button.
3. Complete the GitHub authentication process successfully.
5. Instead, the "Sign In" button still appears, and your profile icon and name are missing.

**Expected Behavior:**
After a successful GitHub sign-in, the user should be shown a profile icon and name displayed.

**Actual Behavior:**
The "Sign In" button still appears. Profile icon and name are not displayed.

**Screenshots/Attachments:**
image

### PR DESCRIPTION
**Pull request by khemchand-twt11**
### Fix : After the user signs in through GitHub, the PR enables the user to access their profile with various options using dropdown.

Fixes #530 

**Before:** 
**Description:** After sign in with github there is no user profile on the home page.

https://github.com/Real-Dev-Squad/website-dashboard/assets/84411728/bab4d521-c0e1-4968-ad9d-5b81e4c16132

**After:** 
**Description:** If the user has logged in he/she will be able to see user profile with dropdown options.

https://github.com/Real-Dev-Squad/website-dashboard/assets/84411728/bd4007de-5672-4923-9271-91d40bbd8a01

**Test Cases**
Alltestcasepassing

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -319,3 +319,85 @@ addClickEventListener(
...
+  {
+    name: 'Identity',
+    link: 'https://my.realdevsquad.com/identity',
+  },
+];
+async function handleUserSignin() {
+  try {
+    const self_user = await getSelfUser();
+    if (self_user) {
+      console.log(self_user);
```
Comment (sahsisunny) [inline script.js:349]:
```suggestion
```

File: script.js [hunk 2]
```diff
@@ -319,3 +319,85 @@ addClickEventListener(
...
+      signInButton.style.display = 'none';
+      const dropdown = document.getElementById('dropdown');
+      const userInfo = document.querySelector('.user-info');
+      const username = document.getElementById('user-name');
+      const userImage = document.getElementById('user-img');
+
+      username.innerText = `Hello, ${self_user.first_name}!`;
+      userImage.setAttribute('src', self_user?.picture.url);
+      userInfo.classList.add('active');
+      const unorderedList = createElement({
```
Comment (sahsisunny) [inline script.js:360]: Could you please provide the accurate name instead of using "unorderedList"?

File: script.js [hunk 3]
```diff
@@ -319,3 +319,85 @@ addClickEventListener(
...
+        type: 'li',
+        classList: ['dropdown-item', 'dropdown-link'],
+        id: 'signout-option',
+      });
+
+      unorderedList.append(signOutElement);
+      signOutElement.innerText = 'Sign Out';
+      dropdown.append(unorderedList);
+
+      //Evnet listener on user-info
```
Comment (sahsisunny) [inline script.js:394]: This comment is important?

File: style.css
Comment (sahsisunny) [inline style.css:1]: Can you use REM instead of PX?

File: utils.js [hunk 4]
```diff
@@ -65,3 +65,15 @@ function debounce(func, delay) {
 async function addDelay(milliSeconds) {
   await new Promise((resolve) => setTimeout(resolve, milliSeconds));
 }
+
+async function signout() {
+  try {
+    await fetch('https://api.realdevsquad.com/auth/signout', {
```
Comment (sahsisunny) [inline utils.js:71]: Could you please use the constants we have for this URL?

File: utils.js [hunk 5]
```diff
@@ -65,3 +65,15 @@ function debounce(func, delay) {
...
+      method: 'GET',
+      credentials: 'include',
+    });
+    location.reload();
+  } catch (error) {
+    console.error('Signout failed:', error);
+  }
```
Comment (sahsisunny) [inline utils.js:78]: We have a utility fetch function that we use to call APIs can you use them?

**Review by sahsisunny — CHANGES_REQUESTED** [review]
Please write the **tests** for the changes and add test stats in the description.

File: utils.js (hunk 4 above)
Comment (khemchand-twt11) [reply utils.js:71]: ok

**Review by shubhamsigdar1 — COMMENTED** [review]
Where is the test??

**Comment by khemchand-twt11** [comment]
> Where is the test??

Didn't know about writing tests.

**Review by Achintya-Chatterjee — COMMENTED** [review]
Solve the merge conflicts

**Review by sahsisunny — DISMISSED** [review]

File: script.js [hunk 6]
```diff
@@ -338,6 +338,102 @@ addClickEventListener(
...
+      userImage.setAttribute('src', self_user?.picture.url);
+      userInfo.classList.add('active');
+      const dropdownList = createElement({
+        type: 'ul',
+        attributes: {
+          class: 'dropdown-list',
+        },
+      });
+
+      DROPDOWN_OPTIONS.map((option) => {
```
Comment (Ajeyakrishna-k) [inline script.js:386]: what's the reason for using map here?

Comment (khemchand-twt11) [reply script.js:386]: Nothing specific. should i change it to forEach ?

File: script.js [hunk 7]
```diff
@@ -338,6 +338,102 @@ addClickEventListener(
...
+          id: 'signout-option',
+        },
+      });
+      signOutElement.classList.add('dropdown-link');
+
+      dropdownList.append(signOutElement);
+      signOutElement.innerText = 'Sign Out';
+      dropdown.append(dropdownList);
+
+      userInfo.addEventListener('click', () => {
```
Comment (iamitprakash) [inline script.js:425]: what about below code

userInfo.addEventListener('click', () => {
  if (dropdown.classList.contains('active')) {
    dropdown.classList.remove('active');
  } else {
    dropdown.classList.add('active');
  }
});

**Review by sahsisunny — APPROVED** [review]

**Review by iamitprakash — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]

**Pull request merged**